code_search/code_search/pipeline.py

```diff
@@ -0,0 +1,91 @@
+import os
+import apache_beam as beam
+import apache_beam.io as io
+from apache_beam.options.pipeline_options import StandardOptions, PipelineOptions, GoogleCloudOptions, SetupOptions
+from apache_beam.io.gcp.internal.clients import bigquery
+
+
+def create_pipeline_opts(args):
+    options = PipelineOptions()
+    options.view_as(StandardOptions).runner = 'DataflowRunner'
+
+    google_cloud_options = options.view_as(GoogleCloudOptions)
+    google_cloud_options.project = args.project
+    google_cloud_options.job_name = args.job_name
+    google_cloud_options.temp_location = '{}/{}/temp'.format(args.storage_bucket, args.job_name)
+    google_cloud_options.staging_location = '{}/{}/staging'.format(args.storage_bucket, args.job_name)
+
+    options.view_as(SetupOptions).setup_file = os.path.join(os.path.dirname(os.path.dirname(__file__)), 'setup.py')
+
+    return options
+
+
+class SplitRepoPath(beam.DoFn):
+    def process(self, element, *args, **kwargs):
+        nwo, path = element.pop('repo_path').split(' ')
+        element['nwo'] = nwo
+        element['path'] = path
+        yield element
+
+
+class TokenizeCodeDocstring(beam.DoFn):
+    def process(self, element, *args, **kwargs):
+        from code_search.tokenizer import get_function_docstring_pairs
```

Comment:
Why is the import here? Is this because of how Dataflow serializes local functions for remote execution?

Reply:
That was exactly the problem. Serialization loses all the references that were brought into scope via `imports`.